docs/pages/docs/getting-started/app-router.mdx

```diff
@@ -148,24 +148,45 @@ export const config = {
 
 ### `app/[locale]/layout.tsx` [#layout]
 
-The `locale` that was matched by the middleware is available via the `locale` param and can be used to configure the document language.
+The `locale` that was matched by the middleware is available via the `locale` param and can be used to configure the document language. Additionally, we can use this place to pass configuration from `i18n.ts` to Client Components via `NextIntlClientProvider`.
 
 ```tsx filename="app/[locale]/layout.tsx"
-export default function LocaleLayout({
+import {getMessages} from 'next-intl/server';
+
+export default async function LocaleLayout({
   children,
   params: {locale}
 }: {
   children: React.ReactNode;
   params: {locale: string};
 }) {
+  // Providing all messages to the client
+  // side is the easiest way to get started
+  const messages = await getMessages();
```

Comment:
This component could also use `useMessages`, but I think we can reasonably suggest implementing `LocaleLayout` as an async Server Component. A minor benefit is that this ensures that `NextIntlClientProvider` is rendered from a Server Component now. This enables automatically inheriting configuration from `i18n.ts`. If the user makes the root layout a Client Component, this wouldn't be possible and more wiring is needed.